stl/inc/xmemory

```diff
@@ -2174,6 +2174,35 @@ _NODISCARD constexpr bool _Allocators_equal(const _Alloc& _Lhs, const _Alloc& _R
     }
 }
 
+#if _HAS_CXX23
+template <class _Ty, class = void>
+inline constexpr bool _Has_member_from_primary = false;
+template <class _Ty>
+inline constexpr bool _Has_member_from_primary<_Ty, void_t<typename _Ty::_From_primary>> = true;
+
+// Avoid using allocate_at_least when the allocator publicly derives from std::allocator:
+// "old" allocators might override allocate and deallocate but fail to override allocate_at_least.
+template <class _Alloc>
+inline constexpr bool _Allocate_at_least_is_safe =

```

Comment:
Looks like not true for `std::allocator`, as it has `_From_primary` too, so apparently `_Allocate_at_least_is_safe` is false for `std::allocator`. Why?

Reply:
Because `allocator::allocate_at_least` currently does nothing useful, it simply returns the result of calling `allocate`. That said, we should have a comment in `allocator::allocate_at_least` pointing out that `_Allocate_at_least_is_safe` will need adjustment if and when we improve `allocate_at_least`.

Reply:
I suppose there's a case to be made that `std::allocator::allocate_at_least` is _safe_ despite that we don't want to use it. Should we rename `_Allocate_at_least_is_safe` to `_Should_allocate_at_least` or something?

Reply:
Yes. Or better yet, start using `std::allocator::allocate_at_least` and avoid the need to rename and a comment.

Reply:
I like `_Should_allocate_at_least`, I'll validate and push changes. I don't like the idea of calling `std::allocator::allocate_at_least` at this time - that's adding an unnecessary function call in debug mode, and asking the inliner to do extra work in release mode, for no user benefit at this time (or for the foreseeable future).